ruoyi-common/src/test/java/com/ruoyi/common/support/ConvertTest.java

```diff
@@ -0,0 +1,205 @@
+
+import org.junit.Test;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.text.NumberFormat;
+import java.text.ParseException;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.TreeSet;
+
+import static com.ruoyi.common.support.Convert.*;
+import static org.junit.Assert.*;
+
+public class ConvertTest {
+
+    @Test
+    public void testToStr() {
+        assertEquals("bar", toStr(null, "bar"));
+        assertEquals("123", toStr(123, "bar"));
+        assertEquals("foo", toStr("foo"));
+    }
+
+    @Test
+    public void testToChar() {
+        assertEquals('b', (char) toChar(null, 'b'));
+        assertEquals('1', (char) toChar(1, 'b'));
+        assertEquals('f', (char) toChar('f'));
+    }
+
+    @Test
+    public void testToByte() {
+        assertEquals(new Byte("8"), toByte(null, (byte) 8));
+        assertEquals(new Byte("1"), toByte((byte) 1, (byte) 8));
+        assertEquals(new Byte("8"), toByte("", (byte) 8));
+        assertEquals(new Byte("123"), toByte("123", (byte) 8));
+        assertEquals(new Byte("8"), toByte("foo", (byte) 8));
+        assertEquals(new Byte("1"), toByte(1));
+    }
+
+    @Test
+    public void testToShort() {
+        assertEquals(new Short("8"), toShort(null, (short) 8));
+        assertEquals(new Short("1"), toShort((short) 1, (short) 8));
+        assertEquals(new Short("8"), toShort("", (short) 8));
+        assertEquals(new Short("123"), toShort("123", (short) 8));
+        assertEquals(new Short("8"), toShort("foo", (short) 8));
+        assertEquals(new Short("1"), toShort(1));
+    }
+
+    @Test
+    public void testToNumber() {
+        assertEquals(8, toNumber(null, 8));
+        assertEquals(8, toNumber("", 8));
+        assertEquals(123L, toNumber("123", 8));
+        assertEquals(8, toNumber("foo", 8));
+        assertEquals(1, toNumber(1));
+    }
+
+    @Test
+    public void testToInt() throws ParseException {
+        assertEquals(new Integer("8"), toInt(null, 8));
+        assertEquals(new Integer("1"), toInt(NumberFormat.getInstance().parse("1"), 8));
+        assertEquals(new Integer("8"), toInt("", 8));
+        assertEquals(new Integer("123"), toInt("123", 8));
+        assertEquals(new Integer("8"), toInt("foo", 8));
+        assertEquals(new Integer("1"), toInt(1));
+    }
+
+    @Test
+    public void testToIntArray() {
+        assertEquals(Arrays.asList(), Arrays.asList(toIntArray(null, "")));
+        assertEquals(Arrays.asList(0, 0, 0, 0), Arrays.asList(toIntArray("a,b,c,d")));
+    }
+
+    @Test
+    public void testToLongArray() {
+        assertEquals(Arrays.asList(), Arrays.asList(toLongArray(null, "")));
+        assertEquals(Arrays.asList(null, null, null, null), Arrays.asList(toLongArray("a,b,c,d")));
+    }
+
+    @Test
+    public void testToStrArray() {
+        assertEquals(Arrays.asList("a", "b", "c", "d"), Arrays.asList(toStrArray("a,b,c,d")));
```

Comment:
Instead of calling `asList` for both, change expected to `new String[] {"a", "b", "c", "d"}`

Reply:
Your right.. not sure why I did it like that